packages/iTwinUI-react/src/core/Dialog/DialogMain.tsx

```diff
@@ -117,13 +132,18 @@ export const DialogMain = React.forwardRef<HTMLDivElement, DialogMainProps>(
             'iui-dialog-default': styleType === 'default',
             'iui-dialog-full-page': styleType === 'fullPage',
             'iui-dialog-visible': isOpen,
+            'iui-dialog-draggable': isDraggable,
```

Comment:
i think this needs a CSS fix. it unnecessarily applies `left` and `top` which impacts positioning when resizing

https://user-images.githubusercontent.com/9084735/185697821-c3d9de81-e57b-4072-94f1-c50dceac1b36.mov

Reply:
But if you don't move the dialog it will still resize according to the screen. So decided that it would do the same after dragging. Users can set concrete size if they wish.

Reply:
> But if you don't move the dialog it will still resize according to the screen. So decided that it would do the same after dragging.

What if we remove the `left`/`right` in CSS and calculate the initial position using JS (only for draggable dialog)? I think it's nicer to preserve the position of draggable dialogs when the containers get resized.

> Users can set concrete size if they wish.

How would this help?

Reply:
Draggable now has static size

Reply:
Because of `left` and `top` it has size what it is. So without `left`, `top`, `max-width`, `max-height` it will span through entire screen.

Reply:
Not sure I understand. The default dialog styles would still supply the max-width and max-height. I'm only suggesting removing the `max-height`/`max-width` from the draggable dialog styles.

Check out the styles I have unchecked in this screenshot. It still looks the same without all these styles.
![](https://user-images.githubusercontent.com/9084735/188235152-0429903e-4797-430d-a6c2-e932b2d914cc.png)

So all that's needed in JS is to calculate the initial `transform` for centering the dialog initially.

Reply:
But you also have to undo `iui-dialog-draggable` changes that I am not sure about. I think those max width and height are intentional @FlyersPh9. I guess I will undo my changes related to this comment tree and fix that later because it needs CSS changes then and I don't want to do so much resetting trough `style`.

Reply:
Lets tackle that in a separate PR then.